### PR TITLE
Message lifecycle

### PR DESCRIPTION
**Pull request by jeffmahoney**
This PR adds some tools to allow external buffer lifecycle handling a little easier.

### CODE REVIEW

**Review by efd6 — COMMENTED** [review]
Please add godoc to the new additions.

File: auparse/auparse.go [hunk 1]
```diff
@@ -210,6 +211,25 @@ func Parse(typ AuditMessageType, message string) (*AuditMessage, error) {
 	return msg, nil
 }
 
+func ParseBytes(typ AuditMessageType, messageBytes []byte, msg *AuditMessage) error {
```
Comment (efd6) [inline auparse/auparse.go:214]: Can we have a test for this?

File: auparse/auparse.go [hunk 2]
```diff
@@ -59,6 +59,7 @@ type AuditMessage struct {
 	data   map[string]string // The key value pairs parsed from the message.
 	tags   []string          // The keys associated with the event (e.g. the values set in rules with -F key=exec).
 	error  error             // Error that occurred while parsing.
+	Owner  interface{}
```
Comment (efd6) [inline auparse/auparse.go:62]: Where/how is this used?

**Comment by jeffmahoney** [comment]
I'm going to drop this one as there's no reason to couple the lifecycles of the RawAuditMessage (and its buffer) and the final AuditMessage, which makes this PR unnecessary.

**Pull request closed without merging**